vault_test.go

```diff
@@ -103,13 +105,18 @@ func TestVaultClient_GetSecret(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			c := tt.cli
 			res, err := c.GetSecret(tt.path)
+			// fmt.Println("Vault Test: ", tt.name)            // ***
+			// fmt.Println("Metadata: ", res.Metadata.Version) // ***
 			if (err != nil) != tt.wantErr {
 				t.Errorf("Client.GetSecret() error = %v, wantErr %v", err, tt.wantErr)
 				return
 			}
 			if !reflect.DeepEqual(res.KV, tt.wantKv) || !reflect.DeepEqual(res.JSONSecret, tt.wantJSON) {
 				t.Errorf("Client.GetSecret() = %v, want %v", res.KV, tt.wantKv)
 			}
+			if (tt.wantMetadataVersion == 0 && res.Metadata.Version != 0) || (tt.wantMetadataVersion != 0 && res.Metadata.Version == 0) {
```

Comment:
These conditions are fragile. I replaced them with:

```go
if tt.wantMetadataVersion != res.Metadata.Version {
```

However, I now see failures in the tests:

```
Testing with Vault version 1.4.1
TestMain: Preparing Vault server
TestMain: Vault Prepared; Running Tests
--- FAIL: TestVaultClient_GetSecret (12.01s)
    --- FAIL: TestVaultClient_GetSecret/kvv2 (0.00s)
        vault_test.go:118: Metadata Version = 9, want 1
    --- FAIL: TestVaultClient_GetSecret/json-secretV2 (0.00s)
        vault_test.go:118: Metadata Version = 9, want 1
FAIL
exit status 1
FAIL	github.com/TrueTickets/vaultlib	17.978s
```

If I re-run the tests, then the Metadata version jumps from 9 to 10. The problem is that the test data is not being cleaned up from Vault.

I'd like @athak 's help on this front. Specifically, the test setup provides a Vault setup via `integration_test.go` and `test-files/initVaultDev.sh` however there is not teardown / cleanup. Does Vault have any commands that we could run to clear the data?

Reply:
Vault is being run in dev mode so there is no persistence. So just by stopping the container, it should be good to go. I agree that the test should shut it down/remove it after the run but I understand why you might want to keep it around to inspect the data in it. I think that should be optional behavior and the default should be to clean up after the run.
I can take of that in separate PR if you want.

Reply:
My bad, there is no container, it's just stopping the process. But running it in a container is also an improvement to be made.

Reply:
Given that the original author has it running as a process, let's keep it as a process and not move to a container.

I think being in a position to opt-out of clean-up makes a lot of sense. However, let's start with the initial feature: we need to clean up the data. What are you proposing achieve this first goal?

Reply:
Atha and I spoke over the phone, and he will be opening a separate PR that enhances the test script with test cleanup.